Changing tests

tsenapathi opened the pull request.

bgruening commented on tools/mdanalysis/rdf.xml:
```diff
@@ -51,10 +51,10 @@
-                <has_n_columns n="2" />
-              </assert_contents>
-             </output>
+             <assert_contents>
```
you could add additional asserts here

bgruening commented:
Thanks @tsenapathi! Its strange that my commits are here as well, can you rebase this branch against master?

KJNaidoo commented:
Excellent,  It looks like this issue is resolved for publication.

bgruening commented on tools/mdanalysis/test-data/RDF_raw_data.tabular:
```diff
@@ -1,100 +0,0 @@
-0.025	0.0
-0.075	0.0
```
@tsenapathi it looks like travis is returning a different result in which the second column is `nan`.
```
+3.825	nan
+3.875	nan
+3.925	nan
+3.975	nan
+4.025	nan
+4.075	nan
+4.125	nan
+4.175	nan
```

These are exactly the kind of error the reviewer wanted to catch I assume. It seems to me like a valid error or a difference in requirements.

tsenapathi replied:
@bgruening @chrisbarnettster is this a problem with travis or the tool? any ideas how to solve this?

chrisbarnettster replied:
@tsenapathi . Assert that it is not nan?

chrisbarnettster replied:
A very simplistic test ...

    <assert_contents>
        <has_n_columns n="2" />
        <not_has_text text="nan" />
    </assert_contents>

chrisbarnettster replied:
For this specific dataset the outcomes are known. Even with minor changes in precision we should be able to match the output using 'has_line_matching '
For example this line "3.275	6336.434022284689" can be matched with 
<has_line_matching expression="3.275\s+6336.43" />.
This is only for ensuring the outcome for this particular test data set.

chrisbarnettster replied:
not sure why nan. 0.0 makes sense because this is such a short trajectory.

chrisbarnettster replied:
see example assert (will need minor adjustment) used here https://github.com/usegalaxy-eu/workflow-testing/pull/40/commits/244fba640e318df5808df403dd5c92e245bb0e4e

bgruening commented:
@tsenapathi let me know when you need a review.

tsenapathi commented:
Hi, @bgruening and @chrisbarnettster ,
See my latest commits to change tests, I have added more assert tests and also cleaned up few more other issues. Please have a look at this.

tsenapathi commented:
@bgruening you can review it now 😃

bgruening commented:
Merci!
